addons/website_slides/wizard/base_partner_merge.py

```diff
@@ -0,0 +1,93 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+
+from odoo import api, models
+
+
+class MergePartnerAutomatic(models.TransientModel):
+    _inherit = 'base.partner.merge.automatic.wizard'
+
+    @api.model
+    def _merge_partner_with_preferences(self, src_partners, dst_partner,
+                                        res_model, uniq_field, preferences, keep_dst_partner=False):
+        """ Merge records of "table" for the given partners following the given preferences.
+
+        ! It is the caller responsibility to check the ACL, this method is executed in sudo.
+        Use this method when the table contains a unique key per partner to select the
+        records to assign to the dst_partner from the record of src_partners and dst_partner.
+        Indeed, when there is an uniq constraint on partner_id and another column we cannot
+        simply assign all record of src_partners to dst_partner without potentially create
+        conflicts.
+        When multiple records with the same unique key per partner among the record of the
+        given partners (src_partners and dst_partner) are encountered, this method choose
+        the one with the highest preference and assign it to dst_partner. Of course, if there
+        is only one, it is just assigned to the dst_partner.
+
+        :param recordset src_partners: source partners
+        :param recordset dst_partner: destination partner
+        :param str res_model: model name
+        :param str uniq_field: column name of the unique key per partner
```

Comment:
If I understood correctly, this methods handle m2m relationships. I would specify it in the docstring and maybe rename a few things like 
```suggestion 
        :param str res_model: many to many relationship model
        :param str other_record_field: field name of the unique "other" record per partner
```